src/main/java/net/snowflake/client/core/QueryContextCache.java

```diff
@@ -152,148 +137,163 @@ public void deserializeFromArrowBase64(String data) {
         // Clear the cache
         clearCache();
 
-        logger.debug("deserializeFromArrowBase64() returns");
+        logger.debug("deserializeQueryContextJson() returns");
         // Log existing cache entries
         logCacheEntries();
 
         return;
       }
 
-      byte[] decoded = Base64.getDecoder().decode(data);
-      ByteArrayInputStream input = new ByteArrayInputStream(decoded);
-      try (ArrowStreamReader reader = new ArrowStreamReader(input, new RootAllocator())) {
-        while (reader.loadNextBatch()) {
-          VectorSchemaRoot root = reader.getVectorSchemaRoot();
-          for (int i = 0; i < root.getRowCount(); ++i) {
-            QueryContextElement qce = deserializeEntry(root, i);
-            // Merge the element in the existing cache
-            merge(qce.id, qce.readTimestamp, qce.priority, qce.context);
+
+
+    try{
+      ObjectMapper mapper = new ObjectMapper();
+      JsonNode rootNode = mapper.readTree(data);
```

Comment:
I don't think this fragment should be manually decoded, only the context should be opaque to the client. The rest is just part of the protocol between the server and the client.

Reply:
@sfc-gh-tcruanes are you suggesting this should be automatically read while deserializing the server response into a `QueryContextDTO` rather than explicitly deserializing it here?

Reply:
yes, I don't understand the need for us to explicitly serialize/deserialize ourself those query context while we are are already exchanging JSON fragment as part of the API. When GS sends back the response, part of the response can be an optional query context (null or absent if none) and some when the client push back the query context on the createQuery API. This can be a DTO or explicit annotation on the QueryContext object.